xarray/backends/common.py

```diff
@@ -234,23 +253,42 @@ def set_variables(self, variables, check_encoding_set,
 
             self.writer.add(source, target)
 
-    def set_necessary_dimensions(self, variable, unlimited_dims=None):
+    def set_dimensions(self, variables, unlimited_dims=None):
```

Comment:
Can you add a docstring or comment explaining what this method does?

This would help new developers (including myself) come onboard with backend development.